apollo-federation/src/sources/connect/json_selection/known_var.rs

```diff
@@ -1,41 +1,28 @@
-use std::borrow::Cow;
+use std::str::FromStr;
 
 #[cfg(test)]
 use super::location::WithRange;
 use crate::sources::connect::variable::Namespace;
-use crate::sources::connect::variable::VariableNamespace;
-use crate::sources::connect::variable::VariablePathPart;
-use crate::sources::connect::variable::VariableReference;
 
 #[derive(PartialEq, Eq, Clone, Hash)]
 pub(crate) enum KnownVariable {
-    This,
-    Args,
-    Config,
-    Status,
+    Identifier(Namespace),
     Dollar,
     AtSign,
```

Comment:
```suggestion
    External(Namespace),
    Dollar,
    AtSign,
```
Could we call this variant something like `::External`, to reflect that these namespace variables are provided from the outside, in contrast to `$` and `@` which always refer internally within the data?

My gripe with `Identifier` is it's representing a variable like `$this` which is intentionally not a valid `Identifier` according to the grammar.